src/mozilla_sec_eia/extract.py

```diff
@@ -0,0 +1,244 @@
+            "run_id"
+        ][0]
+    )
+
+
+def _get_filings_to_extract(
+    experiment_name: str,
+    metadata: pd.DataFrame,
+    continue_run: bool = False,
+    num_filings: int = -1,
+):
+    """Get filings that should be extracted by run.
+
+    Args:
+        experiment_name: Name of mlflow experiment.
+        metadata: Metadata for full set of filings to potentially extract.
+        continue_run: Whether to continue a previous extraction run.
+        num_filings: Number of filings to extract.
+    """
+    extraction_metadata = pd.DataFrame(
+        {"filename": pd.Series(dtype=str), "success": pd.Series(dtype=bool)}
+    ).set_index("filename")
+    extracted = pd.DataFrame()
+    run_id = None
+    if continue_run:
+        most_recent_run = _get_most_recent_run(experiment_name)
+        extraction_metadata = ExtractionMetadataSchema.validate(
+            _load_artifact_as_csv(
+                most_recent_run, "/extraction_metadata.csv"
+            ).set_index("filename")
+        )
+        extracted = _load_artifact_as_csv(most_recent_run, "/extracted.csv")
+        run_id = most_recent_run.info.run_id
+
+    filings_to_extract = metadata[~metadata["filename"].isin(extraction_metadata.index)]
+    if num_filings > 0:
+        filings_to_extract = filings_to_extract.sample(num_filings)
+    return (
+        filings_to_extract,
+        extraction_metadata,
+        extracted,
+        run_id,
+    )
+
+
+def _get_experiment_name(dataset: str, experiment_suffix: str | None = None) -> str:
+    experiment_name = f"{dataset}_extraction"
+    if experiment_suffix is not None:
+        experiment_name += f"_{experiment_suffix}"
+    return experiment_name
+
+
+def compute_validation_metrics(
+    computed_set: pd.DataFrame,
+    validation_set: pd.DataFrame,
+    value_col: str,
+) -> dict:
+    """Compute precision and recall for extraction compared to validation set.
+
+    Arg:
+        computed_set: Extracted data.
+        validation_set: Expected extraction results.
+        value_col: Column to compare when computing metrics.
+    """
+    # Get initial length of both sets
+    computed_len = len(computed_set)
+    validation_len = len(validation_set)
+
+    # Get index of rows only in one set and make Null in other set
+    idx_validation_only = validation_set.index.difference(computed_set.index)
+    padded_compute_set = pd.concat(
+        [
+            computed_set[value_col],
+            pd.Series([None] * len(idx_validation_only), index=idx_validation_only),
+        ]
+    ).sort_index()
+    idx_compute_only = computed_set.index.difference(validation_set.index)
+    padded_validation_set = pd.concat(
+        [
+            validation_set[value_col],
+            pd.Series([None] * len(idx_compute_only), index=idx_compute_only),
+        ]
+    ).sort_index()
+
+    true_positives = (padded_compute_set == padded_validation_set).sum()
```

Comment:
I think it would also be useful to include a measure of the intersection between value counts. Like if the compute set column looks like [B, A, A, A] and the validation set column looks like [A, B, A, A] I think it could be useful to know that they share the same value counts. I'm not sure what the best metric is here. Maybe take `.value_counts()` for `padded_compute_set` and `padded_validation_set` and then take the percent difference between value count sets?

Reply:
Yeah I could definitely see that being useful, particularly for the exhibit 21 extraction. I lean towards implementing a metric like this separately though, because order is pretty important for how I've structured the basic 10k extraction, so I don't think it would be as relevant.